app/models/take_part_page.rb

```diff
@@ -12,6 +12,9 @@ class TakePartPage < ApplicationRecord
 
   mount_uploader :image, ImageUploader, mount_on: :carrierwave_image
 
+  skip_callback :commit, :after, :store_image!
+  after_save :store_image!
+
```

Comment:
Nice work digging into this. In the [commit message](https://github.com/alphagov/whitehall/pull/5253/commits/4c658210ed2d76ac94fe7f72d6721c315ce2aefc) we say that we do our publishing to the Publishing API through an `after_save` - though as far as I can tell we do [this in an after_commit](https://github.com/alphagov/whitehall/blob/4c658210ed2d76ac94fe7f72d6721c315ce2aefc/lib/publishes_to_publishing_api.rb#L6)? So I'm wondering if there's another callback I'm missing or if we maybe have a callback ordering problem? https://github.com/alphagov/whitehall/blob/4c658210ed2d76ac94fe7f72d6721c315ce2aefc/lib/publishes_to_publishing_api.rb#L6

Reply:
Ah, yes, think I was getting confused.  I've updated the commit message to better reflect reality.

I think this is definitely down to ordering, but I cannot see anything incorrect with the ordering as specified in the model.

Reply:
@brucebolt cool - I see what you mean. I had a quick look myself and can see that there is something odd with the after_commit hooks. I did experiment and we can resolve this by re-ordering the file rather than swapping the hook: https://github.com/alphagov/whitehall/commit/0288a45827180adf52b0ef577ab5c29bf90980bc

Reply:
This peculiarity now explains why I couldn't understand the issue initially.  I've updated this commit to move the include of `PublishesToPublishingApi`, which has resolved the issue.

Reply:
@brucebolt super!